packages/flutter/lib/src/widgets/tween_animation_builder.dart

```diff
@@ -212,3 +216,163 @@ class _TweenAnimationBuilderState<T extends Object?>
     return widget.builder(context, _currentTween!.evaluate(animation), widget.child);
   }
 }
+
+/// Widget that animates a [Tween] value and repeats indefinitely.
+///
+/// The animation continuously cycles from [Tween.begin] to [Tween.end].
+/// When [reverse] is true, the animation reverses direction when it reaches
+/// the end. When [paused] is true, the animation stops at its current value.
+///
+/// {@macro flutter.widgets.TweenAnimationBuilder.optimizations}
+///
+/// {@tool dartpad}
+/// This example shows a continuously rotating square that can be paused and resumed.
+///
+/// ** See code in examples/api/lib/widgets/tween_animation_builder/repeating_tween_animation_builder.0.dart **
+/// {@end-tool}
+///
+/// See also:
+///
+///  * [TweenAnimationBuilder], which animates a tween value once.
+///  * [AnimationController.repeat], the underlying mechanism.
+class RepeatingTweenAnimationBuilder<T extends Object> extends StatefulWidget {
+  /// Creates a widget that repeats a tween animation.
+  RepeatingTweenAnimationBuilder({
+    super.key,
+    required this.tween,
```

Comment:
Do you think we can loosen the limitation of `tween` and instead allowing `Animatable` here? Reasoning:
* I expect that this class is mostly used to play looping animations. Currently a smooth loop needs either `reverse: true` or special periodic properties (such as angles). This class can not be used for, for example, a TranslateX that increases ease-in and then decreases ease-in, since the animation is asymmetrical and has the same beginning and end.
* By roughly examining the code I think the only place that use the special properties of `Tween` are using its `begin` and `end` in `shouldChange`.